suricata-ocsf/package.yaml

```diff
@@ -117,150 +98,120 @@ pipelines:
     disabled: false
     definition: |
       // tql2
+      let $rcode_id = {
+        NOERROR: 0,
+        FORMERROR: 1,
+        SERVERROR: 2,
+        NXDOMAIN: 3,
+        NOTIMP: 4,
+        REFUSED: 5,
+        YXDOMAIN: 6,
+        YXRRSET: 7,
+        NXRRSET: 8,
+        NOTAUTH: 9,
+        NOTZONE: 10,
+        DSOTYPENI: 11,
+        BADSIG_VERS: 16,
+        BADKEY: 17,
+        BADTIME: 18,
+        BADMODE: 19,
+        BADNAME: 20,
+        BADALG: 21,
+        BADTRUNC: 22,
+        BADCOOKIE: 23,
+      }
+      let $rcode = {
+        NOERROR: "NoError",
+        FORMERROR: "FormError",
+        SERVERROR: "ServError",
+        NXDOMAIN: "NxDomain",
+        NOTIMP: "NotImp",
+        REFUSED: "Refused",
+        YXDOMAIN: "YXDomain",
+        YXRRSET: "YXRRSet",
+        NXRRSET: "NXRRSet",
+        NOTAUTH: "NotAuth",
+        NOTZONE: "NotZone",
+      }
       subscribe "suricata"
       where @name == "suricata.dns"
-      this = { event: this }
+      this = { suricata: this }
+      // === Classification ===
       // TODO: Proper request/response event splitting.
-      has_query = false
-      has_response = false
-      if event.dns.rrname != null {
+      ocsf.has_query = false
+      ocsf.has_response = false
+      if suricata.dns.rrname != null {
         // Version <=7.0 DNS query logs.
         // See https://docs.suricata.io/en/latest/upgrade/8.0-dns-logging-changes.html
-        query = {
-          hostname: event.dns.rrname,
-          type: event.dns.rrtype,
+        ocsf.query = {
+          hostname: suricata.dns.rrname,
+          type: suricata.dns.rrtype,
         }
-        has_query = true
+      ocsf.has_query = true
       }
-      if event.dns.queries[0].rrname != null {
+      if suricata.dns.queries[0].rrname != null {
```

Comment:
Should this check whether there is at least one query, i.e., `if suricata.dns.queries.length() > 0`?

Reply:
We discussed this in a brief call and came up with this:

```
has_query = false
has_response = false
if dns.has("queries") {
  // Version >=8 DNS query logs.
  // See https://docs.suricata.io/en/latest/upgrade/8.0-dns-logging-changes.html
  ocsf.query = {
    hostname: suricata.dns.queries.first().rrname,
    type: suricata.dns.queries.first().rrtype,
  }
  has_query = true
} else {
  // Version <=7 DNS query logs.
  // See https://docs.suricata.io/en/latest/upgrade/8.0-dns-logging-changes.html
  ocsf.query = {
    hostname: suricata.dns.rrname,
    type: suricata.dns.rrtype,
  }
  has_query = true
}
```